plugin/src/main/java/org/opensearch/ml/model/MLTask.java

```diff
@@ -23,45 +23,161 @@
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
 import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.ml.common.dataset.MLInputDataType;
+import org.opensearch.ml.common.parameter.FunctionName;
 
 @Getter
 @EqualsAndHashCode
-public class MLTask implements Writeable {
-    private final String taskId;
+public class MLTask implements ToXContentObject, Writeable {
```

Comment:
We can move this class to the ml-common if we find it necessary to do so in the next PRs.